### PR TITLE
Display medias of message attachments

### PR DESCRIPTION
**Pull request by koallann**
PR-attachments

### CODE REVIEW

File: fcm-channel/src/main/java/io/fcmchannel/sdk/chat/ChatMessageViewHolder.java [hunk 1]
```diff
@@ -179,6 +196,64 @@ private void setupMessageHourTextColor(TextView hour, boolean incoming) {
...
+            final String url = attachment.getUrl();
+
+            if (AttachmentHelper.isImageUrl(url)) {
+                if (firstImageAttachment == null) {
+                    firstImageAttachment = attachment;
+                }
+                attachmentMedias.add(new ImageMedia(url));
+            }
+            else if (AttachmentHelper.isVideoUrl(url)) {
+                final String thumbnailUri = "https://i.imgur.com/y5Oth3E.png"; // black background
```
Comment (johncordeiro) [inline fcm-channel/src/main/java/io/fcmchannel/sdk/chat/ChatMessageViewHolder.java]: What about adding this constant in the AttachmentHelper class?

File: fcm-channel/src/main/java/io/fcmchannel/sdk/chat/ChatMessageViewHolder.java [hunk 2]
```diff
@@ -104,7 +121,6 @@ void bindView(Message chatMessage) {
         this.chatMessage = chatMessage;
 
         message.setText(chatMessage.getText());
-        Linkify.addLinks(message, Linkify.ALL);
```
Comment (johncordeiro) [inline fcm-channel/src/main/java/io/fcmchannel/sdk/chat/ChatMessageViewHolder.java:107]: The links sent in the chat are still clickable without this line?

File: fcm-channel/src/main/java/io/fcmchannel/sdk/util/AttachmentHelper.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    private static final String REGEX_IMAGE = "(http|https)://.*\\.(jpg|jpeg|png|gif|tif|tiff|bmp)";
+    private static final String REGEX_VIDEO = "(http|https)://.*\\.(mp4|mkv|wmv|avi|flv)";
+
+    public static String getUrlFileExtension(final String url) {
+        final int index = url.lastIndexOf(".") + 1;
+        return index <= url.length() - 1 ? url.substring(index) : "";
+    }
+
+    public static boolean isImageUrl(final String url) {
```
Comment (johncordeiro) [inline fcm-channel/src/main/java/io/fcmchannel/sdk/util/AttachmentHelper.java:22]: Where is the case when the image comes at the end of the text message, especially in push notifications?

File: fcm-channel/src/main/java/io/fcmchannel/sdk/chat/ChatMessageViewHolder.java (hunk 2 above)
Comment (koallann) [reply fcm-channel/src/main/java/io/fcmchannel/sdk/chat/ChatMessageViewHolder.java:107]: I put the attribute `android:autoLink="web"` in the message text view.

File: fcm-channel/src/main/java/io/fcmchannel/sdk/util/AttachmentHelper.java (hunk 3 above)
Comment (koallann) [reply fcm-channel/src/main/java/io/fcmchannel/sdk/util/AttachmentHelper.java:22]: The handling is at `BundleHelper` and the `extractMediaUrls` method from this class is used. That method returns a pair with first element being the text without urls and second with the extracted urls.

Comment (johncordeiro) [reply fcm-channel/src/main/java/io/fcmchannel/sdk/util/AttachmentHelper.java:22]: Perfect

**Review by johncordeiro — APPROVED** [review]

**Pull request merged**